helm/chaos-mesh/templates/ingress.yaml

```diff
@@ -32,17 +33,52 @@ metadata:
     {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
+  {{- if eq (include "chaos-dashboard.ingress.apiVersion" $) "networking.k8s.io/v1" }}  
+  {{- with .Values.dashboard.ingress.ingressClassName }}
+  ingressClassName: {{ . }}
+  {{- end }}
+  {{- end }}
   rules:
-  {{- range .Values.dashboard.ingress.hosts }}
+  {{- if .Values.dashboard.ingress.hosts }}
+    {{- range .Values.dashboard.ingress.hosts }}
     - host: {{ .name | quote }}
       http:
         paths:
-        {{- $paths := default (list "/") .paths }}
         {{- range $paths }}
           - path: {{ . }}
+            {{- if eq (include "chaos-dashboard.ingress.apiVersion" $) "networking.k8s.io/v1" }}
+            pathType: Prefix
+            {{- end }}
+            backend:
+              {{- if eq (include "chaos-dashboard.ingress.apiVersion" $) "networking.k8s.io/v1" }}
+              service:
+                name: {{ template "chaos-dashboard.svc" . }}
+                port:
+                  name: http
+              {{- else }}
+              serviceName: {{ template "chaos-dashboard.svc" . }}
+              servicePort: http
+              {{- end }}
+        {{- end }}
+    {{- end }}
+  {{- else }}
+    - http:
+        paths:
+        {{- range $paths }}
+          - path: {{ . }}
+            {{- if eq (include "chaos-dashboard.ingress.apiVersion" $) "networking.k8s.io/v1" }}
+            pathType: Prefix
+            {{- end }}
             backend:
+              {{- if eq (include "chaos-dashboard.ingress.apiVersion" $) "networking.k8s.io/v1" }}
+              service:
+                name: {{ template "chaos-dashboard.svc" . }}
+                port:
+                  name: http
+              {{- else }}
               serviceName: {{ template "chaos-dashboard.svc" . }}
               servicePort: http
+              {{- end }}
```

Comment:
I don't quit understand the point of this else?

Reply:
I didn't understand what you meant. To explain the above, if `ApiVersion` is not `networking.k8s.io/v1`, it is set to the ingress format of the previous version.

Reply:
Something like this: https://gist.github.com/MarkAndersonTrocme/8d277794708249e10d5fa5bedd08aff3

Removing lines 53-73

Reply:
@KimDoubleB what do you think?

Reply:
extract `eq (include "chaos-dashboard.ingress.apiVersion" $) "networking.k8s.io/v1"` into a vriable is much better not it is not required.

and it seems it is a little hard with the name thing to instead  `apiIsStable` 🙈

Reply:
@MarkAndersonTrocme 

I think your example should have `host` field. Does it work without `host` field?

- ref: https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-rules

Reply:
@KimDoubleB I have the host field in my example on [line 44](https://gist.github.com/MarkAndersonTrocme/8d277794708249e10d5fa5bedd08aff3#L44)

Reply:
But feel free to merge it's just a nitpick